hedera-node/src/main/java/com/hedera/services/store/models/EvmTransaction.java

```diff
@@ -0,0 +1,86 @@
+package com.hedera.services.store.models;
+
+import com.hedera.services.context.TransactionContext;
+import com.hedera.services.context.properties.GlobalDynamicProperties;
+import com.hedera.services.store.contracts.stubs.StubbedBlockchain;
+import com.hederahashgraph.api.proto.java.ContractCallTransactionBody;
+import com.swirlds.common.CommonUtils;
+import org.apache.tuweni.bytes.Bytes;
+import org.hyperledger.besu.crypto.SECPSignature;
+import org.hyperledger.besu.ethereum.chain.Blockchain;
+import org.hyperledger.besu.ethereum.core.Address;
+import org.hyperledger.besu.ethereum.core.Difficulty;
+import org.hyperledger.besu.ethereum.core.Hash;
+import org.hyperledger.besu.ethereum.core.ProcessableBlockHeader;
+import org.hyperledger.besu.ethereum.core.Transaction;
+import org.hyperledger.besu.ethereum.core.Wei;
+
+import java.util.Optional;
+
+import static com.hedera.services.exceptions.ValidationUtils.validateFalse;
+import static com.hedera.services.utils.EntityIdUtils.asSolidityAddressHex;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.CONTRACT_NEGATIVE_GAS;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.CONTRACT_NEGATIVE_VALUE;
+
+public class EvmTransaction {
+	private final GlobalDynamicProperties properties;
+	private final ContractCallTransactionBody transactionBody;
+
+	private long nonce;
+	private Optional<Address> receiverEvmAddress;
+	private Address senderEvmAddress;
+	private SECPSignature signature;
+	private TransactionContext transactionContext;
+
+	public EvmTransaction(GlobalDynamicProperties properties,
+						  ContractCallTransactionBody transactionBody,
+						  TransactionContext transactionContext,
```

Comment:
Given that you are using only the submitting node account from the context, wouldn't it be better to add only itself, rather the whole context?

Reply:
changed